os/mm/kmm_heap/kmm_calloc.c

```diff
@@ -115,9 +122,27 @@ void *kmm_calloc_at(int heap_index, size_t n, size_t elem_size)
 	kheap = kmm_get_heap();
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
 	ARCH_GET_RET_ADDRESS
-	return mm_calloc(&kheap[heap_index], n, elem_size, retaddr);
+	ret = mm_calloc(&kheap[heap_index], n, elem_size, retaddr);
+	if (ret == NULL) {
+#ifdef CONFIG_MM_ASSERT_ON_FAIL
+		lldbg("Allocation failed from %d-th kernel heap.\n", heap_index);
+#else
+		mdbg("Allocation failed from %d-th kernel heap.\n", heap_index);
+#endif
```

Comment:
Let's define below in mm header and use it.
```
mm header
#ifdef CONFIG_MM_ASSERT_ON_FAIL
#define mmfaildbg lldbg    // When CONFIG_MM_ASSERT_ON_FAIL is enabled, we can't use the buffer way like mdbg because of board assert.
#else
#define mmfaildbg mdbg
#endif

xxx.c
mmfaildbg("xxxxx");
```

Reply:
or we can modify debug.h